whiterabbit/src/main/java/org/ohdsi/whiteRabbit/scan/SourceDataScan.java

```diff
@@ -378,10 +399,50 @@ else if (dbType == DbType.BIGQUERY) {
 		return fieldInfos;
 	}
 
+	private List<FieldInfo> processSasFile(String filename) {
+		StringUtilities.outputWithTime("Scanning table " + filename);
+		List<FieldInfo> fieldInfos = new ArrayList<>();
+
+		// TODO: try with resources and print warning on exception
+		FileInputStream inputStream;
+		try {
+			inputStream = new FileInputStream(new File(filename));
+
+			SasFileReader sasFileReader = new SasFileReaderImpl(inputStream);
+
+			// TODO: retrieve more information from the sasFileProperties, like data type and length.
+			SasFileProperties sasFileProperties = sasFileReader.getSasFileProperties();
+			for (Column column : sasFileReader.getColumns()) {
+				fieldInfos.add(new FieldInfo(column.getName()));
+			}
+
+			for (int i = 0; i < sasFileProperties.getRowCount(); i++) {
+				Object[] row = sasFileReader.readNext();
+
+				if (row.length == fieldInfos.size()) { // Else there appears to be a formatting error, so skip
+					for (int j = 0; j < row.length; j++) {
+						fieldInfos.get(j).processValue(row[j] == null ? "" : row[j].toString());
+					}
+				}
+				if (sampleSize != -1 && i == sampleSize)
```

Comment:
Note: sampleSize != -1 is redundant, if sampleSize == -1, then i == sampleSize is always false.